pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/v1/FixedByteChunkSingleValueWriter.java

```diff
@@ -68,37 +68,44 @@ public FixedByteChunkSingleValueWriter(File file, ChunkCompressor compressor, in
 
   @Override
   public void setInt(int row, int value) {
-    _chunkBuffer.putInt(_chunkDataOffset, value);
+    _chunkBuffer.putInt(value);
     _chunkDataOffset += INT_SIZE;
     flushChunkIfNeeded();
   }
 
   @Override
   public void setFloat(int row, float value) {
-    _chunkBuffer.putFloat(_chunkDataOffset, value);
+    _chunkBuffer.putFloat(value);
     _chunkDataOffset += FLOAT_SIZE;
     flushChunkIfNeeded();
   }
 
   @Override
   public void setLong(int row, long value) {
-    _chunkBuffer.putLong(_chunkDataOffset, value);
+    _chunkBuffer.putLong(value);
     _chunkDataOffset += LONG_SIZE;
     flushChunkIfNeeded();
   }
 
   @Override
   public void setDouble(int row, double value) {
-    _chunkBuffer.putDouble(_chunkDataOffset, value);
+    _chunkBuffer.putDouble(value);
     _chunkDataOffset += DOUBLE_SIZE;
     flushChunkIfNeeded();
   }
 
+  @Override
+  protected void writeChunk() {
+    _chunkBuffer.flip();
+    super.writeChunk();
+    _chunkBuffer.clear();
```

Comment:
No need to clear() as it's already cleared in base writeChunk(). Just revert this part and move _chunkBuffer.flip() into base class should be good.